src/MultiplexingDistributedCache/MultiplexingDistributedCache.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Caching.Distributed;
+
+namespace Aqovia.Cache
+{
+    public class MultiplexingDistributedCache : IDistributedCache
+    {
+        private readonly IDistributedCache _primary;
+        private readonly IDistributedCache _secondary;
+        public MultiplexingDistributedCache(IDistributedCache primary, IDistributedCache secondary = default)
+        {
+            _primary = primary;
+            _secondary = secondary;
+        }
+
+        public byte[] Get(string key)
+        {
+            return _primary.Get(key);
+        }
+
+        public async Task<byte[]> GetAsync(string key, CancellationToken token = default)
+        {
+            return await _primary.GetAsync(key, token).ConfigureAwait(false);
+        }
+
+        public void Refresh(string key)
+        {
+            _primary.Refresh(key);
+            _secondary?.Refresh(key);
+        }
+
+        public async Task RefreshAsync(string key, CancellationToken token = default)
+        {
+            await (_secondary?.RefreshAsync(key, token) ?? Task.CompletedTask).ConfigureAwait(false);
```

Comment:
For the synchronous methods we act on the primary cache first but for the async methods we seem to operate on the secondary cache first. Is there any logic to the ordering?

Reply:
No not really. Maybe at first I had a reason, but to be honest I don't have now.

I thought maybe if a failure happens during the action on secondary, it doesn't continue for primary, but then it should be the same for synchronous. So I don't know.